### ISSUE TITLE
GAP OscarInterface / Alnuth tests slow down orthogonal_group(ZLattice)

### ISSUE THREAD

**Issue by benlorenz** [oscar-system/Oscar.jl#2259]
**Describe the bug**
After running the GAP tests for Alnuth (imported in #2220, run via `test/GAP/runtests.jl`) the runtime for some other tests (in `test/Groups/forms.jl`) went up quite a bit:
```
Orthogonal groups of ZZ-lattices |    7      7  6m31.8s
```
before #2220 this took about 2 seconds in the CI (e.g. [here](https://github.com/oscar-system/Oscar.jl/actions/runs/4683316630/jobs/8298266551#step:6:659)).
The six minutes are from here, on master with julia 1.8:
https://github.com/oscar-system/Oscar.jl/actions/runs/4700188792/jobs/8334545934#step:6:670

**To Reproduce**
Copied a few lines together from the test files:
```julia
julia> gram = ZZ[2 1 -1 -1 -1 1 1 -1 0 0 0 0 0 0 0 0; 1 2 -1 -1 -1 1 1 -1 0 0 0 0 0 0 0 0; -1 -1 2 0 1 0 -1 1 0 0 0 0 0 0 0 0; -1 -1 0 2 1 -1 0 0 0 0 0 0 0 0 0 0; -1 -1 1 1 2 0 -1 0 0 0 0 0 0 0 0 0; 1 1 0 -1 0 2 0 -1 0 0 0 0 0 0 0 0; 1 1 -1 0 -1 0 2 -1 0 0 0 0 0 0 0 0; -1 -1 1 0 0 -1 -1 2 0 0 0 0 0 0 0 0; 0 0 0 0 0 0 0 0 2 1 1 0 1 1 1 0; 0 0 0 0 0 0 0 0 1 2 1 0 1 1 0 0; 0 0 0 0 0 0 0 0 1 1 2 0 0 0 1 0; 0 0 0 0 0 0 0 0 0 0 0 2 1 0 -1 0; 0 0 0 0 0 0 0 0 1 1 0 1 4 1 0 1; 0 0 0 0 0 0 0 0 1 1 0 0 1 4 0 0; 0 0 0 0 0 0 0 0 1 0 1 -1 0 0 8 1; 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 18];

julia> L = Zlattice(gram=gram);

julia> @time orthogonal_group(L);
 13.867049 seconds (56.99 M allocations: 2.840 GiB, 6.61% gc time, 99.69% compilation time)

julia> L = Zlattice(gram=gram);

julia> @time orthogonal_group(L);
  0.044626 seconds (315.39 k allocations: 25.195 MiB)

julia> GAP.Globals.TestDirectory(GAP.Globals.DirectoriesPackageLibrary(GAP.Obj("OscarInterface"), GAP.Obj("tst")));                                                                                               Architecture: x86_64-pc-linux-gnu-julia1.8-64-kv8                                                                                                                                                                 testing: /home/datastore/lorenz/software/julia/Oscar-tmp.jl/src/../gap/OscarInterface//tst/alnuth/ALNUT\ H.tst
   35289 ms (719 ms GC) and 11.4MB allocated for alnuth/ALNUTH.tst
testing: /home/datastore/lorenz/software/julia/Oscar-tmp.jl/src/../gap/OscarInterface//tst/alnuth/examp\
les.tst
     541 ms (345 ms GC) and 15.3MB allocated for alnuth/examples.tst
testing: /home/datastore/lorenz/software/julia/Oscar-tmp.jl/src/../gap/OscarInterface//tst/alnuth/manua\
l.tst
     505 ms (364 ms GC) and 3.63MB allocated for alnuth/manual.tst
testing: /home/datastore/lorenz/software/julia/Oscar-tmp.jl/src/../gap/OscarInterface//tst/alnuth/polyn\
ome.tst
   16344 ms (836 ms GC) and 392MB allocated for alnuth/polynome.tst
-----------------------------------
total     52679 ms (2264 ms GC) and 423MB allocated
              0 failures in 4 files

#I  No errors detected while testing

julia> L = Zlattice(gram=gram);    
                                                                       
julia> @time orthogonal_group(L);
175.340849 seconds (1.15 G allocations: 68.066 GiB, 4.45% gc time)
```
Recreating `L` each time is necessary, otherwise the result is fetched from the attribute.

Please also note the almost 70GiB allocations ...

**Expected behavior**
Faster tests!

**System (please complete the following information):**
```julia
julia> Oscar.versioninfo(full=true)
OSCAR version 0.12.1-DEV - #master, 17d5291aa7 -- 2023-04-14 15:12:53 +0200
  combining:
    AbstractAlgebra.jl   v0.29.2
    GAP.jl               v0.9.4
    Hecke.jl             v0.18.8
    Nemo.jl              v0.33.7
    Polymake.jl          v0.9.0
    Singular.jl          v0.18.1
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.5+0
    GAP_jll                 v400.1200.200+0
    Singular_jll            v403.101.500+0
    libpolymake_julia_jll   v0.9.0+0
    libsingular_julia_jll   v0.29.0+0
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 24 × AMD Ryzen 9 5900X 12-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, znver3)
  Threads: 1 on 24 virtual cores
Environment:
  LD_LIBRARY_PATH = /usr/lib64/mpi/gcc/openmpi4/lib64
  JULIA_DEPOT_PATH = /home/datastore/lorenz/software/julia/depot
Official https://julialang.org/ release
```

@fingolfin @ThomasBreuer

**Comment by fingolfin** [comment]
We'll look into it.

**Comment by ThomasBreuer** [comment]
The reason for the problem is the following.
```
julia> GAP.Globals.AssertionLevel()
0

julia> GAP.Globals.TestDirectory(GAP.Globals.DirectoriesPackageLibrary(GAP.Obj("OscarInterface"), GAP.Obj("tst")));
[...]

julia> GAP.Globals.AssertionLevel()
2
```
This bug has been fixed in the GAP master branch with https://github.com/gap-system/gap/pull/5180, thus the problem will disappear with the next version of GAP.
Perhaps the easiest workaround for the moment is to reset the assertion level in `/export/home/sam/julia/packages/Oscar.jl/test/GAP/runtests.jl` after the `GAP.Globals.TestDirectory` call.

(When one looks at the code that actually causes the slowdown, one finds that `ClosureSubgroupNC` contains an `Assert` call that checks whether the closure in question is a subgroup of the parent of the first argument; I would argue that the `NC` in the function name means that no such checks are performed.)

**Comment by fingolfin** [comment]
@ThomasBreuer thanks for the analysis. Can you please make a PR which resets the assertion level as you describe it?

**Comment by fingolfin** [comment]
@ThomasBreuer hmm, it is not clear to me that https://github.com/gap-system/gap/pull/5180 is related? In particular, all the .tst files already use balance START_TEST/STOP_TEST, don't they? 

BTW I also tried to just `Test` each of the `.tst` files separately and all seemed to be fine?

**Comment by ThomasBreuer** [comment]
@fingolfin Yes, using `Test` is safe. However, as I wrote, `TestDirectory` does not call the usual `STOP_TEST` function but replaces it by another function. Before gap-system/gap/pull/5180, this was `function(arg) end`, which does not reset the assertion level. Since gap-system/gap/pull/5180 got merged, this is essentially `STOP_TEST_QUIET`, which takes care of the assertion level. (In this sense, the title of gap-system/gap/pull/5180 is perhaps a bit misleading but not wrong.)
